libsolidity/analysis/DocStringTagParser.cpp

```diff
@@ -202,7 +202,16 @@ bool DocStringTagParser::visit(InlineAssembly const& _assembly)
 				if (valuesSeen.insert(value).second)
 				{
 					if (value == "memory-safe-assembly")
+					{
+						if (_assembly.annotation().markedMemorySafe)
+							m_errorReporter.warning(
```

Comment:
Do we really want the warning now? We wanted library authors to add the comment as early as possible. If it generates a warning, this kind of defeats the purpose.
What about adding a warning starting from 0.9?

Reply:
Ah wait, don't we also have "info" now?

Reply:
The warning is only for
```
/// @solidity memory-safe-assembly
assembly ("memory-safe") {}
```
I.e. doing it both.
That never makes sense - you'd want to use the comment, if you want backwards compatibility or the flag if not.
Maybe I'll rephrase the text.

Reply:
Rephrased.

Reply:
Then again, I could also just remove this warning altogether.

Also should I add an info message about the comment? I'd tend to not saying anything now and making it a warning in 0.9.